clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -51,9 +52,12 @@
     private final ProducerRecord<byte[], byte[]> record1 = new ProducerRecord<>(topic, "key1".getBytes(), "value1".getBytes());
     private final ProducerRecord<byte[], byte[]> record2 = new ProducerRecord<>(topic, "key2".getBytes(), "value2".getBytes());
 
+    @After
+    public void close() {
+        producer.close();
```

Comment:
Yes, that's what I was thinking. Then we can remove the global references and use the same pattern consistently.